controllers/chat/getSignedChannelList.js

```diff
@@ -0,0 +1,73 @@
+const {responseSuccess} = require('../../utils/Responses');
+const GetstreamSingleton = require('../../vendor/getstream/singleton');
+
+const __queryChannelBuilder = async (userId, limit, offset) => {
+  console.log('querying channel', limit, offset);
+  const client = GetstreamSingleton.getChatInstance();
+  const channels = await client.queryChannels({members: {$in: [userId]}}, [{last_message_at: -1}], {
+    limit: limit,
+    offset: offset,
+    state: true
+  });
+  await client.disconnectUser();
+  return channels;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `__queryChannelBuilder` is using a console.log statement which might not be suitable for production code. Consider replacing it with a proper logging mechanism that can be toggled on and off based on the environment (development, production, etc.). Also, this function is disconnecting the user after querying channels. This could lead to unexpected behavior if other operations need to be performed with the same client in the future.

```diff
-  console.log('querying channel', limit, offset);
+  logger.debug(`querying channel with limit: ${limit} and offset: ${offset}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->